### PR TITLE
IGNITE-21331 Sql. Fix deadlock caused by sorting on distributed correlated path

### PR DESCRIPTION
**Pull request by korlov42**
…

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/ColocatedSortAggregatePlannerTest.java [hunk 1]
```diff
@@ -293,47 +295,17 @@ public void noSortAppendingWithCorrectCollation() throws Exception {
...
-                                                .and(input(isTableScan("TEST")))
-                                        ))
-                                ))
-                        ))
-                ),
-                disableRules
-        );
+        RuntimeException e = assertThrows(RuntimeException.class,
+                () -> assertPlan(TestCase.CASE_17, isInstanceOf(IgniteRel.class), disableRules));
+        assertThat(e.getMessage(), containsString("There are not enough rules to produce a node with desired properties"));
```
Comment (AMashenkov) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/ColocatedSortAggregatePlannerTest.java:300]: Maybe, it would be better to disable tests instead of rewriting their code?

Comment (korlov42) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/ColocatedSortAggregatePlannerTest.java:300]: I thought about it, but decided to go with changing the test. First, I want to make sure such plan is impossible, so we need such test anyway. Second, this is not just side effect of my change, but rather intended result. When (or most probably "if") we fix problem with correlated sorted exchange, there is no guarantee the resulting plan will be the same.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java [hunk 2]
```diff
@@ -52,10 +55,26 @@
...
-            "customer", "lineitem", "nation", "orders",
-            "part", "partsupp", "region", "supplier"
-    };
+    /*
+        Minimal configuration of this benchmark requires specifying pathToDataset. Latest known location
+        of dataset is https://github.com/cmu-db/benchbase/tree/main/data/tpch-sf0.01 for scale factor 0.01
+        and https://github.com/cmu-db/benchbase/tree/main/data/tpch-sf0.1 for scale factor 0.1. Dataset
+        is set of CSV files with name `{$tableName}.tbl` per each table and character `|` as separator.
+
+        By default, cluster's work directory will be created as a temporal folder. This implies,
```
Comment (zstan) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java:64]:
```suggestion
        By default, cluster's work directory will be created as a temporary folder. This implies,
```

**Review by zstan — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java (hunk 2 above)
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java]: fixed, thanks!

**Review by AMashenkov — APPROVED** [review]
